### PR TITLE
AK+LibJS: Small fix for Map/Set

### PR DESCRIPTION
**Pull request by davidot**
Also added a bunch of tests to hopefully catch all the behavior (I kinda spread them over a bunch of files)
Fixes all regressions and one new fix of the Map rewrite.

The only thing I'm not sure about is the removal of the optional, feel like I'm missing a case but can't think of any.

### CODE REVIEW

File: Userland/Libraries/LibJS/Tests/builtins/Set/Set.prototype.values.js [hunk 1]
```diff
@@ -12,3 +12,36 @@ test("basic functionality", () => {
...
+        expect(it.next()).toEqual({ value: 1, done: false });
+        expect(it.next()).toEqual({ value: 2, done: false });
+        expect(it.next()).toEqual({ value: 3, done: false });
+        expect(it.next()).toEqual({ value: undefined, done: true });
+        expect(it.next()).toEqual({ value: undefined, done: true });
+        expect(it.next()).toEqual({ value: undefined, done: true });
+    });
+});
+
+test("third appears in entries value when 8 are added in set", () => {
```
Comment (alimpfard) [inline Userland/Libraries/LibJS/Tests/builtins/Set/Set.prototype.values.js:33]: ...what?

Comment (davidot) [reply Userland/Libraries/LibJS/Tests/builtins/Set/Set.prototype.values.js:33]: Oh right this tests isn't needed (it the case where I found the `find_smallest_not_below_iterator` problem

**Comment by linusg** [comment]
`child(eren)` -> `child(ren)`

**Comment by davidot** [comment]
Fixed the commit messages and added a couple of extra tests

**Comment by davidot** [comment]
Found another issue in RedBlackTree `m_minimum` not properly being cleared. Fixed and added another test

**Review by alimpfard — APPROVED** [review]

**Pull request merged**